src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1170,7 +1168,7 @@ private function addSerializerSection(ArrayNodeDefinition $rootNode, callable $e
                     ->{$enableIfStandalone('symfony/serializer', Serializer::class)}()
                     ->children()
                         ->booleanNode('enable_annotations')->end()
-                        ->booleanNode('enable_attributes')->{!class_exists(FullStack::class) ? 'defaultTrue' : 'defaultFalse'}()->end()
+                        ->booleanNode('enable_attributes')->{class_exists(FullStack::class) ? 'defaultFalse' : 'defaultNull'}()->end()
```

Comment:
why defaulting to null ?

Reply:
Attributes should be enabled by default, and this is not the case anymore.

Reply:
To handle the BC layer. If the default to true, then it'll always throw an error with the following config:
```
    serializer:
        enable_annotations: true
```

How should I handle the logic then?

Reply:
See my other comment about the fact that the BC layer should be before normalization (which is also before applying the defaults)